### PR TITLE
CRM-16923 Fix Enotice error by setting payment Processor id

### PR DESCRIPTION
**Pull request by seamuslee001**

### CODE REVIEW

File: CRM/Contribute/Form/Task/PDF.php [hunk 1]
```diff
@@ -182,6 +182,15 @@ public function postProcess() {
       $input['trxn_date'] = isset($contribution->trxn_date) ? $contribution->trxn_date : NULL;
       $input['receipt_update'] = $params['receipt_update'];
       $input['contribution_status_id'] = $contribution->contribution_status_id;
+      $payment_processor_id = NULL;
+      $processorDAO = CRM_Core_DAO::executeQuery("SELECT payment_processor_id
```
Comment (eileenmcnaughton) [inline CRM/Contribute/Form/Task/PDF.php]: If you did "limit 1 " you could use singleValueQuery & save a few lines of code...

File: CRM/Contribute/Form/Task/PDF.php [hunk 2]
```diff
@@ -182,6 +182,12 @@ public function postProcess() {
       $input['trxn_date'] = isset($contribution->trxn_date) ? $contribution->trxn_date : NULL;
       $input['receipt_update'] = $params['receipt_update'];
       $input['contribution_status_id'] = $contribution->contribution_status_id;
+      $payment_processor_id = CRM_Core_DAO::singleValueQuery("SELECT payment_processor_id
+        FROM civicrm_financial_trxn
+        WHERE trxn_id = %1
+        LIMIT 1", array(
+          1 => array($contribution->trxn_id, 'String')));
+      $input['paymentProcessor'] = $payment_processor_id;
```
Comment (eileenmcnaughton) [inline CRM/Contribute/Form/Task/PDF.php]: Probably not worth re-submitting now - but you could assign this to $input['paymentProcessor'] in the line above couldn't you?

**Comment by eileenmcnaughton** [comment]
I've added the merge on pass as Seamus advises he has tested the error goes away & no other appears with this patch. Due to the triviality & readability I think that is enough

**Comment by seamuslee001** [comment]
Test failures are unrelated ping @eileenmcnaughton

**Pull request merged**